dev/com.ibm.ws.cdi.api_fat/fat/src/com/ibm/ws/cdi/api/fat/tests/CDIAPITests.java

```diff
@@ -169,25 +174,26 @@ public void testConversationFilter() throws Exception {
 
     @Test
     @Mode(TestMode.FULL)
-    @ExpectedFFDC("org.jboss.weld.exceptions.DefinitionException")
-    @AllowedFFDC("com.ibm.ws.container.service.state.StateChangeException")
+    @ExpectedFFDC(DEFINITION_EXCEPTION)
+    @AllowedFFDC(STATE_CHANGE_EXCEPTION)
     public void testInjectInjectionPointBeansXML() throws Exception {
         server.setMarkToEndOfLog();
 
         WebArchive injectInjectionPointBeansXMLWar = ShrinkWrap.create(WebArchive.class, INJECT_IP_BEANS_XML_APP_NAME + ".war")
-                                                               .addClass(InjectInjectionPointBeansXMLServlet.class)
-                                                               .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+                                                               .addClass(InjectInjectionPointBeansXMLServlet.class);
+        injectInjectionPointBeansXMLWar = CDIArchiveHelper.addEmptyBeansXML(injectInjectionPointBeansXMLWar);
 
         ShrinkHelper.exportToServer(server, "dropins", injectInjectionPointBeansXMLWar, DeployOptions.SERVER_ONLY);
 
-        String log = server.waitForStringInLog("CWWKZ0002E(?=.*injectInjectionPoint)(?=.*com.ibm.ws.container.service.state.StateChangeException)(?=.*org.jboss.weld.exceptions.DefinitionException)(?=.*WELD-001405)(?=.*BackedAnnotatedField)(?=.*com.ibm.ws.fat.cdi.injectInjectionPointBeansXML.InjectInjectionPointBeansXMLServlet.thisShouldFail)");
+        String log = server.waitForStringInLog(DEFINITION_EXCEPTION_PATTERN_PREFIX
+                                               + InjectInjectionPointBeansXMLServlet.class.getName() + THIS_SHOULD_FAIL_SUFFIX);
```

Comment:
Perhaps a comment on each of these `String log =  ` for what the Exception pattern is expected to resolve to would be a good idea?

Reply:
well I guess it is the long complicated pattern that I was trying to avoid writing out multiple times, which is potentially error prone